### PR TITLE
fix: allow sharing a run instance when using multiprocessing

### PR DESCRIPTION
**Pull request by kptkin**
Description
-----------

- Fixes WB-21080

What does the PR do? Include a concise description of the PR contents.

Fix attach in multiprocessing when sharing a run object across process, it was missing the setting object and hence would raise an exception when trying to access this object

- [x] I updated CHANGELOG.md, or it's not applicable

Testing
-------
How was this PR tested?

enabled the broken functional test

### CODE REVIEW

File: landfill/functional_tests/mp/06-3-share-child-lt-step.py
Comment (dmitryduev) [inline landfill/functional_tests/mp/06-3-share-child-lt-step.py:1]: do you also want to transfer these other files? or do they have limited value or something?

File: wandb/sdk/wandb_init.py [hunk 1]
```diff
@@ -970,7 +970,7 @@ def _attach(
     if run is None:
         run = Run(settings=settings)
     else:
-        run._init()
+        run._init(settings=settings)
```
Comment (dmitryduev) [inline wandb/sdk/wandb_init.py:973]: why is this needed?

**Review by dmitryduev — APPROVED** [review]

File: wandb/sdk/wandb_init.py (hunk 1 above)
Comment (kptkin) [reply wandb/sdk/wandb_init.py:973]: because otherwise the run instance doesn't have a settings object when it is being attached

File: landfill/functional_tests/mp/06-3-share-child-lt-step.py
Comment (kptkin) [reply landfill/functional_tests/mp/06-3-share-child-lt-step.py:1]: no, i removed all of these just left the regular one, i can add another example where the value is less than, but i feel like it is limited value
